apis/core/v1alpha1/clusterobjectdeployment_types.go

```diff
@@ -0,0 +1,59 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterObjectDeploymentSpec defines the desired state of a ClusterObjectDeployment.
+type ClusterObjectDeploymentSpec struct {
+	// Number of old revisions in the form of archived ObjectSets to keep.
+	// +kubebuilder:default=5
+	RevisionHistoryLimit *int `json:"revisionHistoryLimit,omitempty"`
+	// Selector targets ObjectSets managed by this Deployment.
+	Selector metav1.LabelSelector `json:"selector"`
+	// Template to create new ObjectSets from.
+	Template ObjectSetTemplate `json:"template"`
+}
+
+// ClusterObjectDeploymentStatus defines the observed state of a ClusterObjectDeployment
+type ClusterObjectDeploymentStatus struct {
+	// Conditions is a list of status conditions ths object is in.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// DEPRECATED: This field is not part of any API contract
+	// it will go away as soon as kubectl can print conditions!
+	// Human readable status - please use .Conditions from code
```

Comment:
```suggestion
	// Phase is not part of any API contract
	// it will go away as soon as kubectl can print conditions!
	// When evaluating object state in code, use .Conditions instead.
```
Copy/pasta of new Phase comment from the other APIs